packages/shared/src/components/squads/SquadMemberMenu.tsx

```diff
@@ -1,51 +1,158 @@
-import React, { ReactElement, useContext } from 'react';
-import dynamic from 'next/dynamic';
-import { Item } from '@dailydotdev/react-contexify';
+import React, { ReactElement, useContext, useMemo } from 'react';
 import AuthContext from '../../contexts/AuthContext';
 import FlagIcon from '../icons/Flag';
 import { reportSquadMember } from '../../lib/constants';
 import { IconSize } from '../Icon';
+import {
+  SourceMember,
+  SourceMemberRole,
+  SourcePermissions,
+  Squad,
+} from '../../graphql/sources';
+import BlockIcon from '../icons/Block';
+import StarIcon from '../icons/Star';
+import UserIcon from '../icons/User';
+import SquadIcon from '../icons/Squad';
+import { useSquadActions } from '../../hooks/squads/useSquadActions';
+import { usePrompt } from '../../hooks/usePrompt';
+import { UserShortInfo } from '../profile/UserShortInfo';
+import { ModalSize } from '../modals/common/types';
+import { ContextMenu, ContextMenuItemProps } from '../fields/PortalMenu';
 
-const PortalMenu = dynamic(
-  () => import(/* webpackChunkName: "portalMenu" */ '../fields/PortalMenu'),
-  {
-    ssr: false,
-  },
-);
+interface SquadMemberMenuProps {
+  squad: Squad;
+  member: SourceMember;
+}
+
+enum MenuItemTitle {
+  AddAsOwner = 'Add as co-owner',
+  PromoteToModerator = 'Promote to moderator',
+  DemoteToModerator = 'Demote to moderator',
+  DemoteToMember = 'Demote to member',
+  RemoveMember = 'Remove member',
+}
+
+const promptDescription: Record<
+  MenuItemTitle,
+  (memberName: string, squadName: string) => string
+> = {
+  [MenuItemTitle.AddAsOwner]: (memberName, squadName) =>
+    `${memberName} will get the same permissions as you have. Adding as co-owner will not replace you as the owner of ${squadName}`,
+  [MenuItemTitle.PromoteToModerator]: () => '', // TODO: awaiting product's response
+  [MenuItemTitle.DemoteToModerator]: (memberName) =>
+    `${memberName} will no longer have owner permissions. You can always reverse this decision later`,
+  [MenuItemTitle.DemoteToMember]: () => '', // TODO: awaiting product's response
+  [MenuItemTitle.RemoveMember]: (memberName, squadName) =>
+    `${memberName} will be a Blocked members and will no longer have access to ${squadName}. They will not be able to rejoin unless you unblock them.`,
+};
 
 export default function SquadMemberMenu({
```

Comment:
I find this file a bit overwhelming at the moment.
Is there anyway we can clean it out or re-use more parts?

Not sure if it's just me, but just looks a bit messy code-wise.

Reply:
Not sure which parts actually. Can you specify which ones you think requires refactoring?

Reply:
Just in general there is a lot of arrays being structured and think we can clean them up a bit?
Maybe we can have a look together if you want.

Reply:
I mean, those arrays are necessary as those are the options. Sure, let's meet after the web team meeting.

Reply:
I found a way to at least take out the arrays to be coming from a function for easier reading. Hope it helps.